### ISSUE TITLE
segway_description: roslaunch check fails with missing joint_state_publisher dependency

### ISSUE THREAD

**Issue by jack-oquin** [StanleyInnovation/segway_v3#6]
```
$ make run_tests_segway_description_roslaunch-check
Scanning dependencies of target run_tests_segway_description_roslaunch-check_launch
-- run_tests.py: execute commands
  /usr/bin/cmake -E make_directory /home/users/joq/ros/ws/build/test_results/segway_description
  /opt/ros/indigo/share/roslaunch/cmake/../scripts/roslaunch-check -o /home/users/joq/ros/ws/build/test_results/segway_description/roslaunch-check_launch.xml /home/users/joq/ros/ws/src/segway_v3/segway_description/launch 
checking *.launch in directory /home/users/joq/ros/ws/src/segway_v3/segway_description/launch
checking /home/users/joq/ros/ws/src/segway_v3/segway_description/launch/description.launch
Missing package dependencies: segway_description/package.xml: joint_state_publisher
```

**Comment by jack-oquin** [comment]
This diff seems to fix it. I can submit a PR for that, if you like:
```
diff --git a/segway_description/package.xml b/segway_description/package.xml
index ce418ec..e074842 100644
--- a/segway_description/package.xml
+++ b/segway_description/package.xml
@@ -11,6 +11,7 @@
 
   <buildtool_depend>catkin</buildtool_depend>
   <build_depend>roslaunch</build_depend>
+  <run_depend>joint_state_publisher</run_depend>
   <run_depend>robot_state_publisher</run_depend>
   <run_depend>urdf</run_depend>
   <run_depend>xacro</run_depend>
```

**Comment by jack-oquin** [comment]
There's another, similar bug in ``segway_navigation_apps``: ``costmap2d`` is missing.

**Comment by PatrickHussey** [comment]
Thanks for the fixes and letting me know about the issues. I'll make the recommended updates to fix them. Mind me asking why you are running the tests? I would love some help getting this stuff released in the ROS distro if that's something you might be willing to help with.

**Comment by jack-oquin** [comment]
> Mind me asking why you are running the tests? I would love some help getting this stuff released in the ROS distro if that's something you might be willing to help with.

We were running the tests on our own code, see [utexas-bwi/bwi](https://github.com/utexas-bwi/bwi) for the top level. 

Since our [utexas-bwi/segbot](https://github.com/utexas-bwi/segbot) repository depends on ``segway_v3``, which is not released, we have to check out [our modified version of segway_v3](https://github.com/utexas-bwi/segway_v3) in every workspace. (Our modifications are currently in the ``piyushk/remove_ros_ip_dependence`` branch of that fork.)

We would love to have a ROS release of ``segway_v3`` that we could use and depend on. Right now, we need to comment out some of our dependencies and roslaunch checks to keep the ROS build farm and Travis CI from failing. 

I am willing to help you make a release. There are some other problems for which we do not yet have a fix. We would be happy to submit a PR for the fixes we already have.

I'll open another issue for a ``setup.bash`` error involving some absolute workspace path name. We don't have a fix for that, yet.

cc: @piyushk, @maxsvetlik, @barisakgun

**Comment by jack-oquin** [comment]
ref: #7

**Comment by PatrickHussey** [comment]
fixed with commit commit 419ebc13c44b43cb410c51a66e1f995af214487d
